pkg/command/map.go

```diff
@@ -3,9 +3,9 @@
 // Copyright (c) 2021 Rak Laptudirm.
 // Licensed under the MIT license.
 
-package commands
+package command
 
 // A CommandMap is the map type used to
 // connect builtin command functions to
 // their names.
-type CommandMap map[string]func([]string) error
+type CommandMap map[string]Command
```

Comment:
If we create a struct for shell (I think calling it shell for now is fine) then we should put this inside that struct and this on startup would read the commands available through the path variable.